src/mito2/src/read/projection.rs

```diff
@@ -203,21 +230,161 @@ enum BatchIndex {
     Field(usize),
 }
 
+/// Gets a vector with repeated values from specific cache or creates a new one.
+fn repeated_vector_with_cache(
+    data_type: &ConcreteDataType,
+    value: &Value,
+    num_rows: usize,
+    cache_manager: &CacheManager,
+) -> common_recordbatch::error::Result<VectorRef> {
+    if let Some(vector) = cache_manager.get_repeated_vector(value) {
+        // Tries to get the vector from cache manager. If the vector doesn't
+        // have enough length, creates a new one.
+        match vector.len().cmp(&num_rows) {
+            Ordering::Less => (),
+            Ordering::Equal => return Ok(vector),
+            Ordering::Greater => return Ok(vector.slice(0, num_rows)),
+        }
+    }
+
+    // Creates a new one.
+    let vector = new_repeated_vector(data_type, value, num_rows)?;
+    // Updates cache.
+    if vector.len() <= MAX_VECTOR_LENGTH_TO_CACHE {
+        cache_manager.put_repeated_vector(value.clone(), vector.clone());
+    }
+
+    Ok(vector)
+}
+
 /// Returns a vector with repeated values.
 fn new_repeated_vector(
     data_type: &ConcreteDataType,
-    value: ValueRef,
+    value: &Value,
```

Comment:
Why not use ConstantVector?

Reply:
We need to return this vector to datafusion for computing so we can't use constant vectors here.

Reply:
The ConstantVector has `to_arrow_array` function too, it will do the replication. I think we can delay this operation when it's needed. But in this vector memory cache, using ConstantVector can save a lot memory.